### ISSUE TITLE
Support floating versions inside `Build.csproj` in `DotNetClassicTool`

### ISSUE THREAD

**Issue by adutton** [Faithlife/FaithlifeBuild#53]
If I have a `Build.csproj` that contains floating versions for the non-`Faithlife.Build` project, it will not be correctly parsed by `DotNetClassicTool`.  Example `Build.csproj`:
```xml
  <ItemGroup>
    <PackageReference Include="Faithlife.Build" Version="5.*" />
    <PackageReference Include="NUnit.ConsoleRunner" Version="3.*" />
  </ItemGroup>
```
This will result in an error message `Missing restored NuGet package: C:\Users\aaron.dutton\.nuget\packages\nunit.consolerunner\3.*` generated by https://github.com/Faithlife/FaithlifeBuild/blob/fcde9c81f6316c4499d3be2b3ec54c41375de120/src/Faithlife.Build/DotNetClassicTool.cs#L64-L65

The NuGet restore works, but the directory version `3.16.2` that was restored does not match `3.*` and thus the warning.

**Comment by adutton** [comment]
A solution would be to enumerate the directories that match and then pick the latest one by semantic version.  This would work for single version matches as well.

**Comment by adutton** [comment]
Shipped in `5.18.0` in https://github.com/Faithlife/FaithlifeBuild/pull/54
